core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -585,7 +621,12 @@ private void initializeWriter(StoreConfiguration cfg, CacheWriter writer, Initia
       if (writer != null) {
          if (writer instanceof DelegatingCacheWriter)
```

Comment:
@wburns DelegatingCacheWriters are now supported correctly. However, singleton writers are not supported as they are not compatible with shared stores, which transactional stores require.